openqabot/approver.py

```diff
@@ -131,6 +132,49 @@ def is_job_marked_acceptable_for_incident(self, job_id: int, inc: int) -> bool:
             pass
         return False
 
+    @lru_cache(maxsize=512)
+    def was_ok_before(self, failed_job_id: int, inc: int) -> bool:
+        # We need a considerable amount of older jobs, since there could be many failed manual restarts from same day
+        older_jobs = self.client.get_older_jobs(failed_job_id, 20)
```

Comment:
I think we need to add sorting here just to be sure that jobs are ordered in right way

Reply:
I think they do come sorted already, see https://openqa.suse.de/tests/13576926/ajax as example.
Anyway, a follow-up plan is to contribute on openQA an official API for this, instead of stealing the one used for the NextPrevious table (even though is supposed to be pretty stable)